src/PackageExtensionCompat.jl

```diff
@@ -5,39 +5,55 @@ export @require_extensions
 const HAS_NATIVE_EXTENSIONS = isdefined(Base, :get_extension)
 
 @static if !HAS_NATIVE_EXTENSIONS
-    using Requires, TOML
+    using MacroTools, Requires, TOML
 
-    function rewrite_import(str, pkgs)
-        parts = split(strip(str))
-        if length(parts) == 1 || (length(parts) ≥ 2 && parts[2] == "as")
-            if parts[1] ∈ pkgs
-                parts[1] = string("..", parts[1])
+    @static if hasmethod(Base.include, Tuple{Function, Module, String})
+        function _include(mapexpr::Function, m::Module, path::AbstractString)
+            Base.include(mapexpr, m, path)
+        end
+    else
+        function _include(mapexpr::Function, m::Module, path::AbstractString)
+            path = abspath(path)
+            cd(dirname(path)) do
+                str = read(basename(path), String)
+                pos = 1
+                while true
+                    (expr, pos) = Meta.parse(str, pos; raise=false)
+                    expr !== nothing || break
+                    m.eval(mapexpr(expr))
+                end
             end
         end
-        join(parts, " ")
     end
 
-    function rewrite_imports(str, pkgs)
-        parts = split(str, ",")
-        parts = map(part -> rewrite_import(part, pkgs), parts)
-        join(parts, ", ")
+    function rewrite(toplevel::Module, pkgs)
+        Base.Fix1(MacroTools.postwalk, block -> rewrite_block(block, toplevel, pkgs))
```

Comment:
I think this is the only place MacroTools is used? Can you remove it? Trying to keep dependencies as minimal as possible.

Reply:
I'll look into it.